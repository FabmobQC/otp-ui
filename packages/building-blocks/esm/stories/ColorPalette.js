import React from "react";
import styled from "styled-components";
import { getMostReadableTextColor } from "@opentripplanner/core-utils/lib/route";
var ColorBlock = styled.div.withConfig({
  displayName: "ColorPalette__ColorBlock",
  componentId: "sc-fil65c-0"
})(["align-items:center;background-color:", ";color:", ";text-shadow:", ";display:flex;font-family:Arial,Helvetica,sans-serif;height:40px;justify-content:space-between;margin:0;padding:10px;width:300px;&:first-of-type{border-radius:8px 8px 0 0;}&:last-of-type{border-radius:0 0 8px 8px;}"], function (props) {
  return props.hex;
}, function (props) {
  return getMostReadableTextColor(props.hex, "#ffffff");
}, function (props) {
  return getMostReadableTextColor(props.hex, "#ffffff") === "#ffffff" ? "1px 1px 2px black" : "";
});

var ColorPalette = function ColorPalette(_ref) {
  var color = _ref.color;
  var colorArray = Object.entries(color);
  return /*#__PURE__*/React.createElement(React.Fragment, null, colorArray.map(function (hue) {
    return /*#__PURE__*/React.createElement(ColorBlock, {
      key: hue[0],
      hex: hue[1]
    }, /*#__PURE__*/React.createElement("p", null, hue[0]), /*#__PURE__*/React.createElement("p", null, hue[1]));
  }));
};

export default ColorPalette;
//# sourceMappingURL=ColorPalette.js.map