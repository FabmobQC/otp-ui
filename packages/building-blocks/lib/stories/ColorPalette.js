"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _route = require("@opentripplanner/core-utils/lib/route");

const ColorBlock = _styledComponents.default.div.withConfig({
  displayName: "ColorPalette__ColorBlock",
  componentId: "sc-fil65c-0"
})(["align-items:center;background-color:", ";color:", ";text-shadow:", ";display:flex;font-family:Arial,Helvetica,sans-serif;height:40px;justify-content:space-between;margin:0;padding:10px;width:300px;&:first-of-type{border-radius:8px 8px 0 0;}&:last-of-type{border-radius:0 0 8px 8px;}"], props => props.hex, props => (0, _route.getMostReadableTextColor)(props.hex, "#ffffff"), props => (0, _route.getMostReadableTextColor)(props.hex, "#ffffff") === "#ffffff" ? "1px 1px 2px black" : "");

const ColorPalette = ({
  color
}) => {
  const colorArray = Object.entries(color);
  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, colorArray.map(hue => {
    return /*#__PURE__*/_react.default.createElement(ColorBlock, {
      key: hue[0],
      hex: hue[1]
    }, /*#__PURE__*/_react.default.createElement("p", null, hue[0]), /*#__PURE__*/_react.default.createElement("p", null, hue[1]));
  }));
};

var _default = ColorPalette;
exports.default = _default;
//# sourceMappingURL=ColorPalette.js.map