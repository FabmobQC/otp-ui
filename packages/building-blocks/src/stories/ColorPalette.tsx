import React from "react";

import styled from "styled-components";
import { getMostReadableTextColor } from "@opentripplanner/core-utils/lib/route";

interface Hue {
  [key: number]: string;
}

interface Props {
  color: Hue;
}

const ColorBlock = styled.div<{ hex: string }>`
  height: 40px;
  width: 300px;
  background-color: ${props => props.hex};
  color: ${props => getMostReadableTextColor(props.hex, "#ffffff")};
  margin: 0;
  display: flex;
  justify-content: space-between;
  padding: 10px;
`;

const ColorPalette = ({ color }: Props): JSX.Element => {
  const colorArray: Array<Hue> = Object.entries(color);

  return (
    <>
      {colorArray.map(
        (hue: Hue): JSX.Element => {
          return (
            <ColorBlock key={hue[0]} hex={hue[1]}>
              <p>{hue[0]}</p>
              <p>{hue[1]}</p>
            </ColorBlock>
          );
        }
      )}
    </>
  );
};
export default ColorPalette;
