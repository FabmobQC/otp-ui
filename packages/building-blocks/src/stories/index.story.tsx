import React, { ReactElement } from "react";
import ColorPalette from "./ColorPalette";

import grey from "../colors/grey";
import blue from "../colors/blue";

export default {
  title: "Building-Blocks/Colors"
};

export const Grey = (): ReactElement => {
  return <ColorPalette color={grey} />;
};

export const Blue = (): ReactElement => {
  return <ColorPalette color={blue} />;
};
