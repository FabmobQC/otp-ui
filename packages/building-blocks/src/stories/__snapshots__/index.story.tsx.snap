// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Building-Blocks/Colors Blue smoke-test 1`] = `
<div class="ColorPalette__ColorBlock-sc-fil65c-0 kffPrV">
  <p>
    50
  </p>
  <p>
    #E5ECF3
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 fjZNXJ">
  <p>
    100
  </p>
  <p>
    #CCDAE7
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 BUDbo">
  <p>
    200
  </p>
  <p>
    #B2C7DB
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 gdTlRw">
  <p>
    300
  </p>
  <p>
    #99B5CF
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 eVcqvA">
  <p>
    400
  </p>
  <p>
    #7FA2C2
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 bVrRBf">
  <p>
    500
  </p>
  <p>
    #6690B7
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 jOBnRa">
  <p>
    600
  </p>
  <p>
    #4D7EAB
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 qhRUD">
  <p>
    700
  </p>
  <p>
    #336B9E
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 nTorg">
  <p>
    800
  </p>
  <p>
    #1A5992
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 ePvtWX">
  <p>
    900
  </p>
  <p>
    #004686
  </p>
</div>
`;

exports[`Building-Blocks/Colors Grey smoke-test 1`] = `
<div class="ColorPalette__ColorBlock-sc-fil65c-0 ihgIyw">
  <p>
    50
  </p>
  <p>
    #ECECEC
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 ebEIFH">
  <p>
    100
  </p>
  <p>
    #D9D9D9
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 fHGSWM">
  <p>
    200
  </p>
  <p>
    #C5C5C5
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 dLWDtv">
  <p>
    300
  </p>
  <p>
    #B3B3B3
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 fvcImc">
  <p>
    400
  </p>
  <p>
    #9F9F9F
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 iEhEko">
  <p>
    500
  </p>
  <p>
    #8C8C8C
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 kcxUth">
  <p>
    600
  </p>
  <p>
    #797979
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 ojcdX">
  <p>
    700
  </p>
  <p>
    #666666
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 dAGsaR">
  <p>
    800
  </p>
  <p>
    #535353
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 imhyEr">
  <p>
    900
  </p>
  <p>
    #404040
  </p>
</div>
`;

exports[`Building-Blocks/Colors Red smoke-test 1`] = `
<div class="ColorPalette__ColorBlock-sc-fil65c-0 jkujVK">
  <p>
    50
  </p>
  <p>
    #F7E7E6
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 gCnVUi">
  <p>
    100
  </p>
  <p>
    #EFCFCE
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 bMKcsy">
  <p>
    200
  </p>
  <p>
    #E7B6B4
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 hDZgTW">
  <p>
    300
  </p>
  <p>
    #DF9E9C
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 dhsBWR">
  <p>
    400
  </p>
  <p>
    #D78683
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 RfVvB">
  <p>
    500
  </p>
  <p>
    #CF6E6B
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 HARpb">
  <p>
    600
  </p>
  <p>
    #C75652
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 bdGDlH">
  <p>
    700
  </p>
  <p>
    #BF3E3A
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 crSKIw">
  <p>
    800
  </p>
  <p>
    #B72620
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 fYLZnJ">
  <p>
    900
  </p>
  <p>
    #AF0E08
  </p>
</div>
`;
